### ISSUE TITLE
[QUESTION] KVS Client Sample App sending the first signaling message takes up to 10 seconds

### ISSUE THREAD

**Issue by mehmetcy84** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1657]
**Environment**: 
Latest Kinesis repo 1.7.3, built on Windows 10 64 bit.
Note: I had to disable threadpool here for the build to succeed:
-DLWS_WITH_THREADPOOL=0 (in repo\CMake\Dependencies\libwebsockets-CMakeLists.txt)

**Issue**: 
The first signalling message sent from the client takes up to 10 seconds irrespective of its content/size.

-kvsWebRTCClientViewer sample is by default using TrickleICE=enabled.
-onIceCandidateHandler sends a signalling message for each ice candidate found.
-The first signalling message sent from the onIceCandidateHandler takes around 10 seconds, the subsequent messages take about 150 milliseconds each.
-If I disable TrickleICE, then the offer which contains the ICE candidates, being the first signalling message sent, takes around 10 seconds.

What might be the reason? Is there a way to remedy this?
Thanks and best regards,

P.S. I build the same repo on my MacOS (WITHOUT setting the DLWS_WITH_THREADPOOL=0)
and the sample client takes less than a millisecond to send signaling messages.
P.P.S I tried building WITH DLWS_WITH_THREADPOOL=0 on MacOS, but the MacOS sample client can still send all signaling messages within a millisecond.

**Comment by niyatim23** [comment]
Hi @mehmetcy84, can you please use the develop branch to build on Windows now, try running the sample application without any modifications to the SDK and see if this issue still persists?

**Comment by niyatim23** [comment]
Please use `develop` and refer to the `windows-openssl-msvc` CI build for steps to build correctly. Closing this issue due to no response. Feel free to reopen / open a new issue if this persists.

**Comment by mehmetcy84** [comment]
Updated to Master 1.9. Using unmodified Master 1.9 for Windows 10 build.
Using the kvsWebRTCClientViewer.exe against the WebRTC Test Page running as master.

The first signalling message ever sent from the client to the master takes 10 seconds. This is the case if I enabled or disabled useTURN, or TrickleICE. For all 4 combinations for these 2 parameters, independent from whatever signaling message goes first, the first and only the first signalling message being sent always takes 10 seconds, and the following signalling messages, if there are any, takes negligible time in comparison.

Here are all the logs for these cases:
- TrickleICE=TRUE,  UseTURN=TRUE (unmodified default)
- TrickleICE=TRUE,  UseTURN=FALSE
- TrickleICE=FALSE,  UseTURN=TRUE
- TrickleICE=FALSE,  UseTURN=FALSE
[TrickleICE_OFF-TURN_OFF.txt](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/13477334/TrickleICE_OFF-TURN_OFF.txt)
[TrickleICE_OFF-TURN_ON.txt](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/13477335/TrickleICE_OFF-TURN_ON.txt)
[TrickleICE_ON-TURN_OFF.txt](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/13477336/TrickleICE_ON-TURN_OFF.txt)
[TrickleICE_ON-TURN_ON.txt](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/13477337/TrickleICE_ON-TURN_ON.txt)

Here is a portion from the default case:

...
2023-11-27 14:21:25.837 PROFILE iceAgentStartGathering(): [Relay candidates setup time] Time taken: 0 ms
2023-11-27 14:21:25.837 INFO    main(): [KVS Viewer] Completed setting local description
2023-11-27 14:21:25.838 INFO    main(): [KVS Viewer] Offer creation successful
2023-11-27 14:21:25.838 INFO    main(): [KVS Viewer] Generating JSON of session description....
2023-11-27 14:21:25.838 VERBOSE signalingClientSendMessageSync(): Signaling Client Sending Message Sync
2023-11-27 14:21:25.839 DEBUG   sendLwsMessage(): SDP_OFFER
...
2023-11-27 14:21:35.724 INFO    main(): [KVS Viewer] Creating data channel...completed
2023-11-27 14:21:35.724 VERBOSE lwsWssCallbackRoutine(): WSS callback with reason 10
2023-11-27 14:21:35.724 DEBUG   lwsWssCallbackRoutine(): Client is writable
2023-11-27 14:21:35.724 INFO    main(): [KVS Viewer] Data Channel open now...

And this is a portion from the TURN=OFF TrickeICE=OFF case:

...
2023-11-27 15:40:29.387 INFO    main(): [KVS Viewer] Candidate collection completed
2023-11-27 15:40:29.388 INFO    main(): [KVS Viewer] Offer creation successful
2023-11-27 15:40:29.388 INFO    main(): [KVS Viewer] Generating JSON of session description....
2023-11-27 15:40:29.388 VERBOSE signalingClientSendMessageSync(): Signaling Client Sending Message Sync
...
2023-11-27 15:40:39.195 DEBUG   lwsWssCallbackRoutine(): Client is writable
2023-11-27 15:40:39.195 INFO    main(): [KVS Viewer] Creating data channel...completed
2023-11-27 15:40:39.195 INFO    main(): [KVS Viewer] Data Channel open now...
2023-11-27 15:40:39.195 VERBOSE lwsWssCallbackRoutine(): WSS callback with reason 10
2023-11-27 15:40:39.196 DEBUG   lwsWssCallbackRoutine(): Client is writable
